lib/ansible/module_utils/redfish_utils.py

```diff
@@ -1061,6 +1194,72 @@ def get_multi_nic_inventory(self, resource_type):
                                inventory['entries']))
         return dict(ret=ret, entries=entries)
 
+    def get_virtualmedia(self, resource_uri):
+        result = {}
+        virtualmedia_list = []
+        virtualmedia_results = []
+        key = "VirtualMedia"
+        # Get these entries, but does not fail if not found
+        properties = ['Description', 'ConnectedVia', 'Id', 'MediaTypes',
+                      'Image', 'ImageName', 'Name', 'WriteProtected',
+                      'TransferMethod', 'TransferProtocolType']
+
+        response = self.get_request(self.root_uri + resource_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+
+        if key not in data:
+            return {'ret': False, 'msg': "Key %s not found" % key}
+
+        virtualmedia_uri = data[key]["@odata.id"]
+
+        # Get a list of all virtual media and build respective URIs
+        response = self.get_request(self.root_uri + virtualmedia_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+
+        for virtualmedia in data[u'Members']:
+            virtualmedia_list.append(virtualmedia[u'@odata.id'])
+
+        for n in virtualmedia_list:
+            virtualmedia = {}
+            uri = self.root_uri + n
+            response = self.get_request(uri)
+            if response['ret'] is False:
+                return response
+            data = response['data']
+
+            for property in properties:
+                if property in data:
+                    virtualmedia[property] = data[property]
+
+            virtualmedia_results.append(virtualmedia)
+        result["entries"] = virtualmedia_results
+        return result
+
+    def get_multi_virtualmedia(self, resource_type):
+        ret = True
+        entries = []
+
+        #  Given resource_type, use the proper URI
+        if resource_type == 'Systems':
+            resource_uris = self.systems_uris
+        elif resource_type == 'Manager':
+            # put in a list to match what we're doing with systems_uris
+            resource_uris = [self.manager_uri]
+
+        for resource_uri in resource_uris:
+            virtualmedia = self.get_virtualmedia(resource_uri)
+            ret = virtualmedia.pop('ret') and ret
+            if 'entries' in virtualmedia:
+                entries.append(({'resource_uri': resource_uri},
+                               virtualmedia['entries']))
+        return dict(ret=ret, entries=entries)
+
```

Comment:
Since GetVirtualMedia is only a command in the "Manager" category (and not in the "Systems" category), you do not need this special  get_multi_virtualmedia() logic. You can just use the simple wrapper form that calls aggregate().